lib/property_sets/delegator.rb

```diff
@@ -21,10 +21,17 @@ def delegate_to_property_set(setname, mappings)
         raise "Second argument must be a Hash" unless mappings.is_a?(Hash)
 
         mappings.each do |old_attr, new_attr|
+          if ActiveRecord.version >= Gem::Version.new('5.0')
+            attribute old_attr, ActiveModel::Type::Value.new
+          end
           define_method(old_attr) { send(setname).send(new_attr) }
           alias_method "#{old_attr}_before_type_cast", old_attr
           define_method("#{old_attr}?") { send(setname).send("#{new_attr}?") }
-          define_method("#{old_attr}=") { |value| send(setname).send("#{new_attr}=", value) }
+          define_method("#{old_attr}_will_change!"){ attribute_will_change!(old_attr) }
+          define_method("#{old_attr}=") do |value|
```

Comment:
Code that calls `update_attribute` will call this setter. Without the `_will_change!` method the changes are not tracked

Reply:
Updated the implementation to use `super` instead, due to the use of `attribute` higher up in the file for managing this attribute.